variation-commons-mongodb/src/test/java/uk/ac/ebi/eva/commons/mongodb/services/FeatureServiceTest.java

```diff
@@ -0,0 +1,150 @@
+import uk.ac.ebi.eva.commons.core.models.Region;
+import uk.ac.ebi.eva.commons.core.models.VariantType;
+import uk.ac.ebi.eva.commons.mongodb.entities.FeatureCoordinatesMongo;
+import uk.ac.ebi.eva.commons.mongodb.entities.VariantMongo;
+import uk.ac.ebi.eva.commons.mongodb.entities.subdocuments.VariantSourceEntryMongo;
+import uk.ac.ebi.eva.commons.mongodb.configuration.MongoRepositoryTestConfiguration;
+import uk.ac.ebi.eva.commons.mongodb.filter.FilterBuilder;
+import uk.ac.ebi.eva.commons.mongodb.filter.VariantRepositoryFilter;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import static com.lordofthejars.nosqlunit.mongodb.MongoDbRule.MongoDbRuleBuilder.newMongoDbRule;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {MongoRepositoryTestConfiguration.class})
+@UsingDataSet(locations = {
+        "/test-data/features.json"})
+public class FeatureServiceTest {
+
+    @Rule
+    public MongoDbRule mongoDbRule = newMongoDbRule().defaultSpringMongoDb("test-db");
+
+    @Autowired
+    private ApplicationContext applicationContext;
+
+    @Autowired
+    private FeatureService featureService;
+
+    private String GENE_ID_EXISTING = "ENSG00000223972";
+
+    private String GENE_NAME_EXISITNG = "DDX11L1";
+
+    private String GENE_ID_NON_EXISTING = "NonExisitngGeneId";
+
+    private String GENE_NAME_NON_EXISITNG = "NonExisitngGeneName";
+
+    @Test
+    public void testGeneIdorGeneName() {
+        List<FeatureCoordinates> featureCoordinates = findByIdOrNameHelper(GENE_ID_EXISTING,
+                GENE_NAME_EXISITNG);
+        assertEquals(GENE_ID_EXISTING, featureCoordinates.get(0).getId());
+        assertEquals(GENE_NAME_EXISITNG, featureCoordinates.get(0).getName());
+
+        featureCoordinates = findByIdOrNameHelper(GENE_ID_EXISTING, null);
+        assertEquals(GENE_ID_EXISTING, featureCoordinates.get(0).getId());
+        assertEquals(GENE_NAME_EXISITNG, featureCoordinates.get(0).getName());
+
+        featureCoordinates = findByIdOrNameHelper(null, GENE_NAME_EXISITNG);
+        assertEquals(GENE_ID_EXISTING, featureCoordinates.get(0).getId());
+        assertEquals(GENE_NAME_EXISITNG, featureCoordinates.get(0).getName());
+
+    }
+
+    private List<FeatureCoordinates> findByIdOrNameHelper(String id, String name) {
+        return featureService.findByIdOrName(id, name);
+    }
+
+    @Test
+    public void testGeneIdorGeneNameNonExisting() {
+        List<FeatureCoordinates> featureCoordinates = findByIdOrNameHelper(GENE_ID_NON_EXISTING,
+                GENE_NAME_NON_EXISITNG);
+        assertTrue(featureCoordinates.size() == 0);
```

Comment:
if this test ever fails, the test logs won't show the size of the list, making harder to understand the problem
```suggestion
        assertEquals(0, featureCoordinates.size());
```